Lib/test/datetimetester.py

```diff
@@ -3122,6 +3122,18 @@ def dst(self, dt): return 1
         with self.assertRaises(TypeError):
             dt_broken.astimezone()
 
+        dt_big = self.theclass(9999, 12, 31, 23, 59, 59)
```

Comment:
I don't love that the original style violates the one-assert-per-test style, and we should probably not continue that.

Let's move this into its own method, like `test_astimezone_max` and `test_astimezone_min`. We can test the values near `datetime.min` and `datetime.max` separately, and use `self.subTest` to try a few different values for time zone.

I'm not sure if `@support.run_with_tz` can be parameterized, but if it can we should try it on a few different time zones.